R/add_rcsi.R

```diff
@@ -0,0 +1,90 @@
+#'
+#' @param data dataset
+#' @param rCSILessQlty Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to rely on less preferred and less expensive food to cope with a lack of food or money to buy it?
+#' @param rCSIBorrow   Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to borrow food or rely on help from a relative or friend to cope with a lack of food or money to buy it?
+#' @param rCSIMealSize Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to limit portion size of meals at meal times to cope with a lack of food or money to buy it?
+#' @param rCSIMealAdult Column representing question- During the last 7 days, were there days (and, if so, how many) when your household had to restrict consumption by adults in order for small children to eat to cope with a lack of food or money to buy it?
+#' @param rCSIMealNb Column representing question - During the last 7 days, were there days (and, if so, how many) when your household had to reduce number of meals eaten in a day to cope with a lack of food or money to buy it?
+#' @param new_colname The prefix for the new columns. It has to be a string.
+#' @return
+#' A dataset with one additional column.
+#' @export
+#' @importFrom rlang :=
+#' @examples
+#' test_data <- data.frame(
+#' rCSILessQlty = c(1,2,3,1),
+#' rCSIBorrow = c(0,0,3,0),
+#' rCSIMealSize = c(4,2,6,1),
+#' rCSIMealAdult = c(4,3,5,0),
+#' rCSIMealNb = c(2,5,NA_integer_,1)
+#' )
+#' add_rcsi(test_data)
+
+
+
+add_rcsi <- function(data,
+                     rCSILessQlty = "rCSILessQlty",
+                     rCSIBorrow = "rCSIBorrow" ,
+                     rCSIMealSize = "rCSIMealSize",
+                     rCSIMealAdult = "rCSIMealAdult",
+                     rCSIMealNb = "rCSIMealNb",
+                     new_colname = "rcsi"
+){
+
+  score_name <- paste0(new_colname, "_score")
+  cat_name <- paste0(new_colname, "_cat")
+  score1 <- paste0(new_colname, "_1")
+  score2 <- paste0(new_colname, "_2")
+  score3 <- paste0(new_colname, "_3")
+  score4 <- paste0(new_colname, "_4")
+  score5 <- paste0(new_colname, "_5")
+
+  rowsum_col <- c(score1,score2,score3,score4,score5)
+  all_newly_created <- c(rowsum_col,score_name,cat_name)
+
+  if(any(all_newly_created %in% names(data))) {
+    print(all_newly_created[all_newly_created %in% names(data)])
+    warning("The above(s) column are existing in the dataset. These columns will be replaced by the function. Please change `new_colname` parameter to keep them.")
+}
+
+  all_names <- c(rCSILessQlty,rCSIBorrow,rCSIMealSize,rCSIMealAdult,rCSIMealNb)
+
+  if(!all(all_names %in%  names(data))){
+    message(all_names[!all_names %in% names(data)])
+    stop("The above column(s) can not be found in the dataset.")
+  }
+
+  class <- sapply(data[all_names],is.numeric)
+
+  if(!all(class==T)){
```

Comment:
Looks good, if possible can also check if safe to coerce to numeric, and auto convert to numeric if so, there is something called varhandle::check.numeric I've seen can do this.

Reply:
Auto conversion can be done but i am afraid it there is any character entry then it will make it NA. To reduce the source of error, i will rather stop the user to use the function and force him to check the dataset. 
<img width="247" alt="image" src="https://github.com/impact-initiatives/addindicators/assets/54301198/56ab75b1-5a91-4b79-a36d-af25bf3ba745">